services/spar/schema/src/V17.hs

```diff
@@ -0,0 +1,33 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module V17
+  ( migration,
+  )
+where
+
+import Cassandra.Schema
+import Imports
+import Text.RawString.QQ
+
+migration :: Migration
+migration = Migration 17 "Remove table `scim_external_ids` (from db migration V10, deprecated in favor of `scim_external`, emptied in `/services/spar/migrate-data/src/Spar/DataMigration/V1_ExternalIds.hs`)" $ do
```

Comment:
```suggestion
migration = Migration 17 "Remove table `scim_external_ids` (from db migration V10, deprecated in favor of `scim_external`, data migrated in `/services/spar/migrate-data/src/Spar/DataMigration/V1_ExternalIds.hs`)" $ do
```
It was not emptied.

Reply:
I wonder if we need to delete the data migration as it will now start failing for new deployments. Perhaps the integration tests are failing for this reason?

Reply:
doesn't look related:

```
FAILED RELEASES:
NAME
test-17k9og6y2c5e-wire-server-2
test-17k9og6y2c5e-wire-server
in /tmp/build/06eff510/wire-server/hack/helmfile.yaml: 2 errors:
err 0: failed processing release test-17k9og6y2c5e-wire-server-2: command "/sbin/helm" exited with non-zero status:

PATH:
  /sbin/helm

ARGS:
  0: helm (4 bytes)
  1: upgrade (7 bytes)
  2: --install (9 bytes)
  3: --reset-values (14 bytes)
  4: test-17k9og6y2c5e-wire-server-2 (31 bytes)
  5: ../.local/charts/wire-server (28 bytes)
  6: --devel (7 bytes)
  7: --wait (6 bytes)
  8: --timeout (9 bytes)
  9: 600s (4 bytes)
  10: --create-namespace (18 bytes)
  11: --namespace (11 bytes)
  12: test-17k9og6y2c5e-fed2 (22 bytes)
  13: --values (8 bytes)
  14: /tmp/helmfile2987696773/test-17k9og6y2c5e-fed2-test-17k9og6y2c5e-wire-server-2-values-59d9c84db8 (96 bytes)
  15: --values (8 bytes)
  16: /tmp/helmfile80875716/test-17k9og6y2c5e-fed2-test-17k9og6y2c5e-wire-server-2-values-77b6595bf7 (94 bytes)
  17: --set (5 bytes)
  18: brig.config.optSettings.setFederationDomain=federation-test-helper.test-17k9og6y2c5e-fed2.svc.cluster.local (107 bytes)
  19: --set (5 bytes)
  20: galley.config.settings.federationDomain=federation-test-helper.test-17k9og6y2c5e-fed2.svc.cluster.local (103 bytes)
  21: --set (5 bytes)
  22: cargohold.config.settings.federationDomain=federation-test-helper.test-17k9og6y2c5e-fed2.svc.cluster.local (106 bytes)
  23: --set (5 bytes)
  24: brig.config.optSettings.setFederationDomainConfigs[0].domain=federation-test-helper.test-17k9og6y2c5e.svc.cluster.local (119 bytes)
  25: --history-max (13 bytes)
  26: 10 (2 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: failed post-install: job failed: BackoffLimitExceeded

COMBINED OUTPUT:
  Release "test-17k9og6y2c5e-wire-server-2" does not exist. Installing it now.
  Error: failed post-install: job failed: BackoffLimitExceeded
err 1: failed processing release test-17k9og6y2c5e-wire-server: command "/sbin/helm" exited with non-zero status:

PATH:
  /sbin/helm

ARGS:
  0: helm (4 bytes)
  1: upgrade (7 bytes)
  2: --install (9 bytes)
  3: --reset-values (14 bytes)
  4: test-17k9og6y2c5e-wire-server (29 bytes)
  5: ../.local/charts/wire-server (28 bytes)
  6: --devel (7 bytes)
  7: --wait (6 bytes)
  8: --timeout (9 bytes)
  9: 600s (4 bytes)
  10: --create-namespace (18 bytes)
  11: --namespace (11 bytes)
  12: test-17k9og6y2c5e (17 bytes)
  13: --values (8 bytes)
  14: /tmp/helmfile2259553671/test-17k9og6y2c5e-test-17k9og6y2c5e-wire-server-values-6456d9d8c9 (89 bytes)
  15: --values (8 bytes)
  16: /tmp/helmfile3658437968/test-17k9og6y2c5e-test-17k9og6y2c5e-wire-server-values-7674cb5c56 (89 bytes)
  17: --set (5 bytes)
  18: brig.config.optSettings.setFederationDomain=federation-test-helper.test-17k9og6y2c5e.svc.cluster.local (102 bytes)
  19: --set (5 bytes)
  20: galley.config.settings.federationDomain=federation-test-helper.test-17k9og6y2c5e.svc.cluster.local (98 bytes)
  21: --set (5 bytes)
  22: cargohold.config.settings.federationDomain=federation-test-helper.test-17k9og6y2c5e.svc.cluster.local (101 bytes)
  23: --set (5 bytes)
  24: brig.config.optSettings.setFederationDomainConfigs[0].domain=federation-test-helper.test-17k9og6y2c5e-fed2.svc.cluster.local (124 bytes)
  25: --history-max (13 bytes)
  26: 10 (2 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: failed post-install: job failed: BackoffLimitExceeded

COMBINED OUTPUT:
  Release "test-17k9og6y2c5e-wire-server" does not exist. Installing it now.
  Error: failed post-install: job failed: BackoffLimitExceeded
make: *** [Makefile:391: kube-integration-setup] Error 1
make: Leaving directory '/tmp/build/06eff510/wire-server'
```

Reply:
if that's really the problem, we may want to do something like this:

```
modified   services/spar/migrate-data/src/Spar/DataMigration/Types.hs
@@ -28,6 +28,8 @@ import qualified System.Logger as Logger
 
 data Migration = Migration
   { version :: MigrationVersion,
+    minSchemaVersion :: Maybe Int,
+    maxSchemaVersion :: Maybe Int,
     text :: Text,
     action :: Env -> IO ()
   }
```

...  and skip migrations with a warning if the schema version doesn't fit.

Anyway it doesn't really make sense to create an empty database, then migrate the schema to the newest version, and then run database migrations from versions that have never been populated.  In this special case we could just manually set a migration version somewhere before migrations are run.

But the more general case is giving me a headache, and makes me think that we should just empty the table and keep it?